crates/editor/src/element.rs

```diff
@@ -5085,6 +5094,8 @@ impl Element for EditorElement {
                         .buffer_rows(start_row)
                         .take((start_row..end_row).len())
                         .collect::<Vec<_>>();
+                    let is_row_soft_wrapped =
+                        |row| buffer_rows.get(row).copied().flatten().is_none();
```

Comment:
This is where we actually use `buffer_rows` to figure out whether we've soft wrapped or not.